nx_cugraph/convert_matrix.py

```diff
@@ -276,3 +284,95 @@ def from_scipy_sparse_array(
     if inplace:
         return create_using._become(G)
     return G
+
+
+@networkx_algorithm(version_added="25.06", fallback=True)
+def to_numpy_array(
+    G,
+    nodelist=None,
+    dtype=None,
+    order=None,
+    multigraph_weight=sum,
+    weight="weight",
+    nonedge=0.0,
+):
+    """MultiGraphs are not yet supported"""
+    # print(" ==> hello!! dispatched to nxcg!")
+    G = _to_graph(G, weight, 1, _get_float_dtype(dtype))
+
+    if nodelist is None:
+        nodelist = list(G)
+
+    N = len(nodelist)
+
+    # use set to check for nodes not in the graph or duplicate nodes
+    nodelist_as_set = set(nodelist)
+    if nodelist_as_set - set(G):
+        raise nx.NetworkXError(
+            f"Nodes {nodelist_as_set - set(G)} in nodelist is not in G"
+        )
+    if len(nodelist_as_set) < N:
+        raise nx.NetworkXError(f"Nodelist {nodelist} contains duplicates")
+
+    # Construct array of fill_value matching resulting shape
+    A = cp.full((N, N), fill_value=nonedge, dtype=dtype, order=order)
```

Comment:
I learned that cupy doesn't handle structured arrays (a.k.a. record arrays) very well, so we may want to use numpy directly in some cases, such as doing:
```suggestion
    use_numpy = dtype.names is not None
    if not use_numpy:
        try:
            A = cp.full((N, N), fill_value=nonedge, dtype=dtype, order=order)
        except Exception:
            use_numpy = True
    if use_numpy:
        A = np.full((N, N), fill_value=nonedge, dtype=dtype, order=order)
```
and use `use_numpy` as appropriate later in the function.